test/firebase_auth_mocks_test.dart

```diff
@@ -516,16 +516,47 @@ void main() {
     expect(decodedToken1['user_id'], decodedToken2['user_id']);
   });
 
-  test('Each edecoded token\'s user_id should unique', () async {
+  test('Each decoded token\'s user_id should unique', () async {
     final user1 = MockUser();
     final user2 = MockUser();
     final idToken1 = await user1.getIdToken();
     final decodedToken1 = JwtDecoder.decode(idToken1);
     final idToken2 = await user2.getIdToken();
     final decodedToken2 = JwtDecoder.decode(idToken2);
-    expect(decodedToken1['user_id'] is String, true);
-    expect(decodedToken2['user_id'] is String, true);
-    expect(decodedToken1['user_id'] != decodedToken2['user_id'], true);
+    expect(decodedToken1['user_id'], isA<String>());
+    expect(decodedToken2['user_id'], isA<String>());
+    expect(decodedToken1['user_id'], isNot(decodedToken2['user_id']));
+  });
+
+  test(
+      'Each decoded token\'s auth_time and exp should be the time in seconds since unix epoch',
+      () async {
+    final idToken = await tUser.getIdToken();
+    final decodedToken = JwtDecoder.decode(idToken);
+    bool isTimeInSecondsSinceUnixEpoch(dynamic number) {
+      if (number is! int) return false;
+      if (number.toString().length != 10) return false;
+      return true;
+    }
+
+    expect(decodedToken['auth_time'], isTimeInSecondsSinceUnixEpoch);
```

Comment:
I using predicate function to wrap my isTimeInSecondsSinceUnixEpoch function, and I found it can improve the error message.
The original error message:
```
Expected: satisfies function
  Actual: <1665109807>

package:test_api                          expect
main.<fn>
```

After wrap by predicate:
```
Expected: is time in seconds since unix epoch
  Actual: <1665110237>

package:test_api                          expect
main.<fn>
```

Reply:
Very nice! Thank you